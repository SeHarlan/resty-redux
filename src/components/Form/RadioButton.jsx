import React from 'react';
import PropTypes from 'prop-types';

const RadioButton = ({ value, selectedOption, onOptionChange }) => (
  <>
    <input checked={selectedOption === value} id={value} type="radio" name="method" value={value} onChange={onOptionChange} />
    <label htmlFor={value}>{value}</label>
  </>
);
RadioButton.propTypes = {
  selectedOption: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  onOptionChange: PropTypes.func.isRequired
};
export default RadioButton;
