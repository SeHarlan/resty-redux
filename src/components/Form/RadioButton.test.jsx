import React from 'react';
import { shallow } from 'enzyme';
import RadioButton from './RadioButton';

describe('RadioButton component', () => {
  it('renders RadioButton', () => {
    const wrapper = shallow(<RadioButton 
      value="GET"
      selectedOption="GET"
      onOptionChange={()=>{}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
