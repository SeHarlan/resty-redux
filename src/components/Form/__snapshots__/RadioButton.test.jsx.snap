// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton component renders RadioButton 1`] = `
<Fragment>
  <input
    checked={true}
    id="GET"
    name="method"
    onChange={[Function]}
    type="radio"
    value="GET"
  />
  <label
    htmlFor="GET"
  >
    GET
  </label>
</Fragment>
`;
